modules/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageServiceTests.java

```diff
@@ -450,6 +450,71 @@ public void testInvalidSettingsRetryConfigurationForLocationModeWithSecondaryFal
         }
     }
 
+    public void testCreateClientWithEndpoints() throws IOException {
+        final Settings settings = Settings.builder()
+            .setSecureSettings(buildSecureSettings())
+            .put("azure.client.azure1.endpoint", "https://account1.zone.azure.net")
+
+            .put("azure.client.azure2.endpoint", "https://account2.zone.azure.net")
+            .put("azure.client.azure2.secondary_endpoint", "https://account2-secondary.zone.azure.net")
+            .build();
+        try (AzureRepositoryPlugin plugin = pluginWithSettingsValidation(settings)) {
+            final AzureStorageService azureStorageService = plugin.azureStoreService.get();
+
+            expectThrows(IllegalArgumentException.class, () -> azureStorageService.client("azure1", LocationMode.PRIMARY_THEN_SECONDARY));
+            expectThrows(IllegalArgumentException.class, () -> azureStorageService.client("azure1", LocationMode.SECONDARY_ONLY));
+            expectThrows(IllegalArgumentException.class, () -> azureStorageService.client("azure1", LocationMode.SECONDARY_THEN_PRIMARY));
+
+            AzureBlobServiceClient client1 = azureStorageService.client("azure1", LocationMode.PRIMARY_ONLY);
+            assertThat(client1.getSyncClient().getAccountUrl(), equalTo("https://account1.zone.azure.net"));
+
+            {
+                AzureBlobServiceClient client = azureStorageService.client(
```

Comment:
Nit: I think we can avoid declaring the `client` variable and declaring a code block to void scoping issues.
```java
assertThat(
    azureStorageService.client("azure2", randomBoolean() ? LocationMode.PRIMARY_ONLY : LocationMode.PRIMARY_THEN_SECONDARY)
        .getSyncClient()
        .getAccountUrl(),
    equalTo("https://account2.zone.azure.net")
);
```